src/members/Members.jsx

```diff
@@ -1,21 +1,93 @@
-import React from "react";
+import React, { useState, useContext, useEffect } from "react";
+import { Table, Button } from "react-bootstrap";
+import { Route } from "react-router-dom"
+import { AuthContext } from "../AuthContext";
+import { BASE_API } from "../config";
+import { SERVICE_UNAVAILABLE_ERROR } from "../messages";
+import MemberProfile from "./MemberProfile";
 
 export default function Members() {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [members, setMembers] = useState([]);
+  const { access_token } = useContext(AuthContext);
+
+  const requestMembersList = {
+    method: "GET",
+    headers: {
+      "Authorization": `Bearer ${access_token}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  useEffect(() => {
+    fetch(`${BASE_API}/users`, requestMembersList)
+      .then(async response => {
+        const data = await response.json();
+        if (response.ok) {
+          return setMembers(data);
+        }
+        setErrorMessage(data.message);
+      })
+      .catch(error =>
+        setErrorMessage(SERVICE_UNAVAILABLE_ERROR)
+      )
+
+  }, [] );
+ 
+  const handleClick = ({member}) => {
     return (
-        <div className="container-fluid" id="members">
-          <div className="top">
-            <h1>
-              This is the Public page for Members list
-            </h1>
-          </div>
-          <div className="middle">
-            <h2>The site is currently under construction...</h2>
-          </div>
-          <div className="bottom">
-            <h3>
-                Thank you for your patience and understanding.
-            </h3>
+      <Route render={() => {
+          return (
+            <div>
+              <MemberProfile member={member} />
+            </div>
+          )
+      }} />
+      )}
+
+  return errorMessage ?
+    <div className="container-fluid" id="memberList">
+      <div className="top">
+        <h1>
+          {errorMessage}
+        </h1>
+      </div>
+    </div>
+    :
+    <>
+      <div className="container" id="membersList">
+        <div className="row mb-5">
+          <div className="col-lg-12 text-center">
+            <h1 className="mt-5">Members List</h1>
           </div>
         </div>
-    )
+        <Table striped bordered hover>
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Skills</th>
+              <th>Need Mentoring</th>
+              <th>Available To Mentor</th>
+              <th>Programs Completed</th>
+              <th>Current Availability</th>
+              <th>Location</th>
+            </tr>
+          </thead>
+          <tbody>
+            {members.map((member) => (
+              <tr key={member.id}>
+                <td><Button variant="link" onClick={handleClick({member})}>{member.name}</Button></td>
```

Comment:
It's actually a link not a button. Why are we using Button? 

Also, onClick expects a function call you, if you see what happens when you click on a member, there's a warning in the console about how onClick is receiving an object. You can fix it by doing this - `onClick={() => handleClick(member)}`

Reply:
I've replaced it with anchor tag <a> which I initially had and fix onclick as your suggestion. But it still doesn't work.